### ISSUE TITLE
ECDSA Example

### ISSUE THREAD

**Issue by paulhovey** [danni/python-pkcs11#54]
Hi, I'm just starting out with this library, so I apologize if I've missed something blatantly obvious!

My environment is Windows running against SoftHSMv2.

I was trying to use the [example for ECDSA](https://python-pkcs11.readthedocs.io/en/latest/index.html#ecdsa) , and I was getting an error that `prime256v1` is not in the dictionary.  It looks as though this key was removed in the asn1crypto library a while ago ( [changes](https://github.com/wbond/asn1crypto/commit/f937f48708d530f8031770ecddb9024e20013c4c#diff-2957980838d0da5b1c4a86dd1950d55f) ), is the example code out of date?  I tried replacing `prime256v1` with `secp256r1` and got past that line of code but failed out with other errors, so I wasn't sure what the correct solution was.  I was able to run the RSA example fine, so my setup seems reasonably correct.

**Comment by danni** [comment]
What was the error you got? Is your version of SoftHSMv2 compiled with EC support?

**Comment by paulhovey** [comment]
Your questions led me to find that my end goal of EDDSA won't work correctly on Windows:  [https://github.com/disig/SoftHSM2-for-Windows/issues/10](https://github.com/disig/SoftHSM2-for-Windows/issues/10)

Regardless, when the ECDSA example is run, I get this error:
```
C:\Data\tmp>python pkcs11_ecdsa.py
Traceback (most recent call last):
  File "pkcs11_ecdsa.py", line 16, in <module>
    pkcs11.Attribute.EC_PARAMS: pkcs11.util.ec.encode_named_curve_parameters('prime256v1'),
  File "C:\Python3\lib\site-packages\pkcs11\util\ec.py", line 31, in encode_named_curve_parameters
    value=NamedCurve.unmap(oid),
  File "C:\Python3\lib\site-packages\asn1crypto\core.py", line 2827, in unmap
    value
ValueError: asn1crypto.keys.NamedCurve._map does not contain an entry for "prime256v1"
```

I think this issue appears regardless of the SoftHSMv2 functionality on Windows, based on the link in the first comment showing where "prime256v1" was replaced by "secp256r1" in the asn1crypto library.

**Comment by danni** [comment]
The name of the curve isn't so important as the OID. Use another name for the same curve or use the OID of the curve (dotted identifier number), that OID is just sent to the HSM. The HSM might reject the mechanism if you don't have EC enabled, or it might reject the parameters if it doesn't have that curve.

**Comment by langbeck** [comment]
I'm also having this problem on Linux with SoftHSM2 2.6.1.
SoftHSM2 module does seems to support EC since I'm able to run:
```bash
pkcs11-tool --module /usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so -l --keypairgen --key-type EC:prime256v1
```

Any thoughts?

**Comment by langbeck** [comment]
I found the issue: the package `python3-asn1crypto` (I've version  `1.4.0`) doesn't have a named curve `prime256v1` anymore (don't know since when), but it does have the original curve to which `prime256v1` is an alias for: `secp256r1`

I think a quick fix would be just to update the docs to use the original `secp256r1` name.

**Comment by danni** [comment]
A pull request would be most welcome.

**Comment by paulhovey** [comment]
Closing issue since it appears @langbeck solved it in #94 , thanks!  :smile:
